### PR TITLE
update: add retry logic when retrieving falco versions

### PR DESCRIPTION
**Pull request by loresuso**
Signed-off-by: Lorenzo Susini <susinilorenzo1@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> /kind flaky-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area library

> /area cli

> /area tests

> /area examples

**What this PR does / why we need it**:
Since `falcoctl` can be run as sidecar container or in a systems unit, ensure that Falco has time to start up and expose the versions endpoint. This PR add a simple retry logic with exponential backoff using a custom http.RoundTripper.
It is possible to debug what is happening using the verbose flag.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by leogr** [comment]
/hold

as discussed privately, we are going to review a bit the backoff logic

File: internal/artifact/follow/follow.go [hunk 1]
```diff
@@ -261,3 +270,33 @@ func (o *artifactFollowOptions) retrieveFalcoVersions(ctx context.Context) error
...
+	for v.attempts < v.MaxRetry {
+		v.attempts++
+		v.Printer.Verbosef("retrieving Falco versions, attempt %d\n", v.attempts)
+		resp, err = v.Base.RoundTrip(req)
+		if err != nil {
+			v.Printer.Verbosef("error: %s. Trying again in %s", err.Error(), v.sleep.String())
+			time.Sleep(v.sleep)
+			v.sleep *= 2
+		} else {
+			return resp, err
```
Comment (FedeDP) [inline internal/artifact/follow/follow.go:297]: I'd just break there, to have a single return statement

File: internal/artifact/follow/follow.go [hunk 2]
```diff
@@ -261,3 +270,33 @@ func (o *artifactFollowOptions) retrieveFalcoVersions(ctx context.Context) error
...
+
+	v.sleep = 1 * time.Second
+
+	for v.attempts < v.MaxRetry {
+		v.attempts++
+		v.Printer.Verbosef("retrieving Falco versions, attempt %d\n", v.attempts)
+		resp, err = v.Base.RoundTrip(req)
+		if err != nil {
+			v.Printer.Verbosef("error: %s. Trying again in %s", err.Error(), v.sleep.String())
+			time.Sleep(v.sleep)
```
Comment (FedeDP) [inline internal/artifact/follow/follow.go:294]: What i don't like about this solutions is that we end up sleeping `2<<5` before actually breaking and returning the error.

**Comment by loresuso** [comment]
@leogr @FedeDP 
I have modified again this PR. Could you please check again? thanks!

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**